pkg/dao/repositories.go

```diff
@@ -43,30 +46,46 @@ func (p repositoryDaoImpl) List() (error, []Repository) {
 		return result.Error, repos
 	}
 	for i := 0; i < len(dbRepos); i++ {
-		repos = append(repos, Repository{
-			UUID:     dbRepos[i].UUID,
-			URL:      dbRepos[i].URL,
-			Revision: dbRepos[i].Revision,
-		})
+		repos = append(repos, modelToInternal(dbRepos[i]))
 	}
 	return nil, repos
 }
 
 func (p repositoryDaoImpl) Update(repoIn Repository) error {
-	var repo models.Repository
+	repo := internalToModel(repoIn)
 
-	result := p.db.Where("uuid = ?", repoIn.UUID).First(&repo)
+	result := p.db.Model(&repo).Updates(repo.MapForUpdate())
```

Comment:
When I run `make repos-import` I see in the database the repositories are public; after run `./releases/external-repos introspect URL` I get for that repository the Public flag is updated to false into the database; I have tried to modify `MapForUpdate` to comment the line for the `Public` field, and it makes it works, for that situation, but that change will avoid the field can be updated.

Reply:
:question: is it accurate to say that once the repository is created (Repository record into the database), the indicator Public can not be changed? if the response is yes, then the above could works

Reply:
This should be fixed now. Previously, the public field was being lost in the Update function and then defaulting to `false`.